### DISCUSSION TITLE
Battery performance - Offline installation - WebCam config

### DISCUSSION THREAD

**Discussion by misterblack0101** [AndresGarciaSobrado91/MSI-Modern15-Hackintosh#8] [Q&A] [answered]
How much is the battery backup, as compared to windows?
Also, is it stable for daily use?

**Comment by AndresGarciaSobrado91** [comment, marked as answer]
Hi! Do you mean how long does the battery last? It's working great, I can't compare it with windows because I didn't load windows in it.
It's stable for daily use. I will update the repo soon with new OC and kexts versions.

**Reply by misterblack0101** [reply]
Hi, @AndresGarciaSobrado91. I hope you're doing well. I'm sorry for the late reply, but after the last try, I was disappointed and switched back to Linux. I need macOs now, for development, and just have a few questions.
1. How do you control the fans?
2. Does sleep and hibernation work?
3. How's the ram management and overall performance?
4. Is there anything that doesn't work?
5. Could you tell me the approx battery life(in hrs).

P.S. I have a similar model, with a 14 inch screen. Thank you for all your help.

**Comment by AndresGarciaSobrado91** [comment]
Yeah, sure:

1. You can control temps and fans with istat menus or HWMonitor
![image](https://user-images.githubusercontent.com/48637183/137205582-1c9ab109-7b3e-41d6-856b-963a971dd9d9.png)
![image](https://user-images.githubusercontent.com/48637183/137205680-3b9f3fe8-8983-4f71-8532-7849d090e07d.png)

2. Working perfect

3. Working perfect, no issues found.

4. Led mute (not really important)

5. 3,5 to 4 Hs
![image](https://user-images.githubusercontent.com/48637183/137206083-4959e4d5-c692-45e0-9fc7-27883d5195ea.png)

**Comment by AndresGarciaSobrado91** [comment]
Also, check https://github.com/lgs3137/MSIFanControl

PS: I use the laptop for the development of Android and iOS apps. So far so good.

**Reply by misterblack0101** [reply]
Hey, was your wifi working when installing?
The last time  I installed, I installed via wifi.
This time, I ain't getting an option for that.
I tried usb tethering, on both Catalina and big sur.
There's no ethernet port on my laptop, so, is there any other way I could install this?

**Comment by AndresGarciaSobrado91** [comment]
The wifi won't work since it's an unsupported card and you need a kext+sofware (Heliport) that can not be loaded at install time. You need to create a USB installer with a full **offline** macOS iso

**Reply by misterblack0101** [reply]
alright, I'll try a full offline installer.
I just finished making my friend a hackintosh, I'll try to create an offline installer from it. thanks 😄

**Reply by misterblack0101** [reply]
Hey @AndresGarciaSobrado91 , I followed this [offline hackintosh installer guide](https://www.hackintoshshop.com/2810/hackintosh-big-sur-guide/), but the installer didn't detect my internal ssd. The disk utility was detecting my ssd, and I formatted it to apfs, but the installer wasn't detecting it.
I tried creating the bootable from windows, as well as macOS, but got no success.
Could you refer me to some other offline installer guide?  Thanks!

**Reply by AndresGarciaSobrado91** [reply]
Maybe is not the installation process itself. What is your SSD brand/model?

**Reply by misterblack0101** [reply]
@AndresGarciaSobrado91 
I was able to install macOs Catalina using [this guide](https://www.reddit.com/r/hackintosh/comments/jrrhox/how_to_make_a_full_offline_installer_for_macos_on/?utm_source=amp&utm_medium=&utm_content=post_body).
I then managed to successfully create and install big sur using dortanias guide via mac.
I'm facing two issues:
1. I'm unable to get my camera to work. The os is not detecting any camera/video device in the system report.
2. How do I get rid of the dual wifi icons (heliport and original wifi) - I only want the heliport one.

Also, Thanks Alot for the EFI, its a lifesaver.

**Comment by AndresGarciaSobrado91** [comment]
Congrats @misterblack0101 ! 

1. If you are using my EFI folder it should work. Try pressing `F6` or `fn key + F6` and then open the app photo Booth to try it.
2. Go to System preferences > Network > Select Wi-Fi tab > Disable `Show Wi-Fi status in menu bar`

**Reply by misterblack0101** [reply]
@AndresGarciaSobrado91 2. worked perfectly.
For the camera, it didn't work. My camera isn't being detected by the system. I tried live boot on linux, and made sure that it was working,, before booting back into macOS, still it didn't detect it.
![image](https://user-images.githubusercontent.com/71558331/139447869-414309b0-8bcd-4a35-8288-2755f1b44081.png)

**Comment by AndresGarciaSobrado91** [comment]
So, in linux you can enable and disable the webcam with `fn + F6` ?

**Reply by misterblack0101** [reply]
No need of function. By default the keys take the utilities, like brightness, volume, camera.
When pressing function, I can use F1, F2..

**Reply by AndresGarciaSobrado91** [reply]
ok but you can enable and disable webcam with F6 right?

**Reply by misterblack0101** [reply]
Yes sir.

**Reply by AndresGarciaSobrado91** [reply]
ok then, you will need to:

1. Remove my USB map kext from the EFI folder and from the config.plist
2. Add USBInjectAll kext (make sure it is added in config.plist too)
3. Reboot system
4. Try to enable web cam with `F6`

If that works, you will need to create a new USB map kext for your machine

**Reply by misterblack0101** [reply]
@AndresGarciaSobrado91 Thanks alot. That worked. once everything was working, I made USBmap from the usbmap app, and put it in place. Now everything's working. Thanks alot for the help 😁
